lib/dshell.py

```diff
@@ -697,14 +699,27 @@ def TCP(self, addr, tcp, ts, **kwargs):
 
         try:
             # close connection
-            if tcp.flags & (dpkt.tcp.TH_FIN | dpkt.tcp.TH_RST):
-                conn = self.find(addr)
+            conn = self.find(addr)
+            if tcp.flags & (dpkt.tcp.TH_FIN | dpkt.tcp.TH_RST) and conn:
+                conn.closeIP(addr[0]) #track if FIN has been seen in connection
+
+            if conn:
+                if conn.connectionClosed(): #if FIN has been seen from both sides, trigger old logic
```

Comment:
Do we still need the self.clientclosed and self.serverclosed attached to the decoder object? It looks like all they're doing is standing in for the conn.connectionClosed() function at this point.